src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.AddProperty.cs

```diff
@@ -80,14 +82,13 @@ internal JsonPropertyInfo CreateProperty(Type declaredPropertyType, Type runtime
             return jsonInfo;
         }
 
-        internal JsonPropertyInfo CreatePolymorphicProperty(JsonPropertyInfo property, Type runtimePropertyType, JsonSerializerOptions options)
+        internal JsonPropertyInfo CreateRootObject(JsonSerializerOptions options)
         {
-            if (property == null)
-            {
-                // Used with root objects which are not really a property.
-                return CreateProperty(runtimePropertyType, runtimePropertyType, null, runtimePropertyType, options);
-            }
+            return CreateProperty(Type, Type, null, Type, options);
```

Comment:
Please add a comment (or retain the existing one): `// Used with root objects which are not really a property.`

It still seems odd to me that we are passing the same argument multiple times here.

Reply:
> It still seems odd to me that we are passing the same argument multiple times here

I'm not sure what you mean?